noxfile.py

```diff
@@ -0,0 +1,60 @@
+import nox
+
+source_files = ("httpx", "tests", "setup.py", "noxfile.py")
+
+
+@nox.session(reuse_venv=True)
+def lint(session):
+    session.install("autoflake", "black", "flake8", "isort", "seed-isort-config")
+
+    session.run("autoflake", "--in-place", "--recursive", *source_files)
+    session.run("seed-isort-config", "--application-directories=httpx")
+    session.run(
+        "isort",
+        "--project=httpx",
+        "--multi-line=3",
+        "--trailing-comma",
+        "--force-grid-wrap=0",
+        "--combine-as",
+        "--line-width=88",
+        "--recursive",
+        "--apply",
+        *source_files,
+    )
+    session.run("black", "--target-version=py36", *source_files)
+
+    check(session)
+
+
+@nox.session(reuse_venv=True)
+def check(session):
+    session.install("black", "flake8", "flake8-bugbear", "mypy")
```

Comment:
Shell we also add `flake8-comprehensions` here?

Reply:
Probably, let's do that in a separate PR though, this one's gotten out of hand.